src/account/PluginManagerInternals.sol

```diff
@@ -138,32 +133,21 @@ abstract contract PluginManagerInternals is IPluginManager {
         );
     }
 
-    function _installPlugin(address plugin, bytes32 manifestHash, bytes memory pluginInstallData) internal {
+    function _installPlugin(address plugin, PluginManifest calldata manifest, bytes memory pluginInstallData)
+        internal
+    {
         AccountStorage storage _storage = getAccountStorage();
 
         if (plugin == address(0)) {
             revert NullPlugin();
         }
 
-        // Check if the plugin exists.
-        if (_storage.pluginManifestHashes.contains(plugin)) {
-            revert PluginAlreadyInstalled(plugin);
-        }
-
+        // TODO: do we need this check? Or switch to a non-165 checking function?
```

Comment:
IMO it's kinda useless, it's maybe a good footgun to prevent some accidents, but in the bad actor case, anyone can just support the interfaces this checks. 

I'm not sure we even need this check at all, what are some potential issues we could be preventing?

Reply:
It's just a footgun check, to validate that the address you're trying to install is in fact a module. We've been using ERC-165, alternatives have created custom, module specific view functions for this purpose.

Reply:
I guess evading undefined behaviour is a good reason, but ultimately, I'm not sure how the risk VS gas tradeoff is. Off the top of my head the only case where this would be helpful is when you're using an app that thinks your account is compatible with some other form of module-- but we do have forced uninstalls which can help here.

Security-wise, any contract can support this check, even an account with added execution functions I guess, so I don't think that's a good reason to keep it. It would prevent accidentally installing an EOA though.

What are your general thoughts? Do we need to explicitly prevent installing non-modules? Or are the cases where this would happen quite rare?